### DISCUSSION TITLE
How to intercept TCP traffics of desktop app on Windows machine using INTRANET ?

### DISCUSSION THREAD

**Discussion by YangShaoyue** [mitmproxy/mitmproxy#5826] [Q&A]
#### Problem Description
Hi, I'm new to mitmproxy, I want to intercept my desktop app TCP traffics running on Windows 11 machine (Client), the APP sends requests to another Windows 10 machine (Server). They communicate with each other under INTRANET or Internal network. Mitmproxy is installed on the client machine.
I also use wireshark to intercept this app TCP traffics, it looks like this:
![图片](https://user-images.githubusercontent.com/12428824/209633487-79fdbe19-137a-43e8-bc11-ad592865a69e.png)

I tried mitmproxy normal mode or wireguard mode, I run `mitmproxy` for normal mode, the windows system proxy setting is below:
![图片](https://user-images.githubusercontent.com/12428824/209633087-3338e66f-8f61-416e-976f-2dbb70deef82.png)

this can intercept HTTP(S) traffics sent by such as Firefox browser:
![图片](https://user-images.githubusercontent.com/12428824/209634052-99919f10-6117-4d60-b001-174ef98ec1d6.png)
but none of TCP traffics is captured.

For wireguard mode `mitmweb --mode wireguard`, regardless of  turnning on/off Windows system proxy setting, there is nothing show in mitmproxy console.

Is there anything I missed? Thank you guys.

#### System Information
![图片](https://user-images.githubusercontent.com/12428824/209635299-9e4b1595-1457-438a-aed8-ce7707b39759.png)

**Comment by rosydawn6** [comment]
What is the full command line being used to start mitmproxy?

Based on some experimentation I have found the following.  The TCP Events methods https://docs.mitmproxy.org/stable/api/events.html will not fire for tcp traffic without the relevant flag from the documentation for those hosts for which tcp level traffic is requested: https://docs.mitmproxy.org/stable/concepts-options/.  See here for some additional context: https://github.com/mitmproxy/mitmproxy/discussions/5510.  
  
` --tcp-hosts ".*"`  (this will capture all)
or 
` --tcp-hosts "internal-host-name-here"`

**Reply by YangShaoyue** [reply]
Thanks for your reply, the full command line being used was `mitmproxy`.
And I tried `tcp_hosts` option, `mitmproxy --tcp-hosts ".*"`, it did show some TCP traffics of my Firefox browser:
 
![图片](https://user-images.githubusercontent.com/12428824/209892025-c37a3296-e538-4dc9-80f3-2b4248514bd8.png)

but none of the APP's TCP traffics was captured. I guess maybe this is because that the desktop app is an internal application of my company, and it uses INTRANET(internal network), not internet.

**Reply by rosydawn6** [reply]
Maybe a host => vpn => mitmproxy link would solve this issue

**Comment by YangShaoyue** [comment]
@mhils Hi, sorry to bother you, do you have any suggestion about this issue ?

**Comment by mhils** [comment]
This looks like your client application is bypassing/not using the system proxy. You need to get your client to use the proxy somehow, or wait until we support Windows OS Proxy mode (hopefully soon). Intranet or not should not matter.

**Reply by YangShaoyue** [reply]
Thanks for your quick reply.
I also think it is very likely that the client app is bypassing system proxy, but I don't have any control of this client app. I have tried some other way according to @rosydawn6 's reply, but there isn't any progress so far.
